alphacsc/utils/dictionary.py

```diff
@@ -92,8 +92,33 @@ def _patch_reconstruction_error(X, z, D):
                    for diff_i in diff], axis=1)
 
 
-def get_lambda_max(X, D_hat, sample_weights=None):
+def get_lambda_max(X, D_hat, sample_weights=None, q=1):
+    """For each atom, compute the associated lambda_max as the q-quantile of
+    the energy patches over the signal length.
```

Comment:
```suggestion
    """For each atom, compute the regularization parameter scaling.
    
    This value is usually defined as the smallest value for which 0 is
    a solution of the optimization problem.
    In order to avoid spurious values, this quantity can also be estimated
    as the q-quantile of the correlation between signal patches and the
    atom.
```